blocks/library/gallery/gallery-image.js

```diff
@@ -75,8 +75,8 @@ class GalleryImage extends Component {
 		}
 	}
 
-	componentWillReceiveProps( { isSelected, image, url } ) {
-		if ( image && ! url ) {
+	componentWillReceiveProps( { isSelected, image } ) {
+		if ( image ) {
```

Comment:
Won't this cause issues on subsequent re-renders of the component where `image` is assigned, but `url` is as well? In other words, do we risk an infinite loop of `setAttributes` -> `componentWillReceiveProps` -> `setAttributes`?

Reply:
Hi @aduth, I don't think so, because of this condition in the with select `image: id && ! url ? getMedia( id ) : null,`, it guarantees that as soon as we have an URL the image will become null so no setAttributes function will be called again.

Reply:
Hi @aduth, thank you for the review, you are right, it is not easy to understand this logic. I did some changes in order to try to make the logic easier.